src/components/form/ContourForm.tsx

```diff
@@ -0,0 +1,242 @@
+            value={formData?.fqdn}
+            onChange={(value) => {
+              updateFormData((prev) => ({ ...prev, fqdn: value }));
+            }}
+          />
+          <div className="help-block">{t('hostname_http_proxies')}</div>
+        </FormGroup>
+        <div className="pf-c-form__group ">
+          {formData?.routes?.map((route, index) => (
+            <div key={index} className="pf-u-mb-md">
+              <RouteForm
+                route={route}
+                onChange={(updatedRoute) => updateRoute(index, updatedRoute)}
+                onDelete={
+                  formData?.routes.length > 1
+                    ? () => removeRoute(index)
+                    : undefined
+                }
+                availableServices={services?.map((svc) => svc.metadata.name)}
+                availablePorts={services?.flatMap((svc) =>
+                  svc.spec.ports?.map((port) => port.port.toString()),
+                )}
+                availableSecrets={secrets?.map(
+                  (secret) => secret.metadata.name,
+                )}
+              />
+            </div>
+          ))}
+          <Button variant="link" onClick={addRoute} className="pf-u-mt-md">
+            <span className="fa fa-plus-circle pf-u-mr-xs"></span>
+            {t('add_route')}
+          </Button>
+        </div>
+        <div>
+          <div>
+            <FormGroup fieldId="secure_route" label={t('security')}>
+              <Switch
+                id="secure-route"
+                label={t('secure_route')}
+                isChecked={formData?.conditional?.secureRoute}
+                onChange={(checked) =>
+                  updateFormData((prev) => ({
+                    ...prev,
+                    conditional: {
+                      ...prev.conditional,
+                      secureRoute: checked,
+                    },
+                  }))
+                }
+              />
+            </FormGroup>
+
+            {formData?.conditional?.secureRoute && (
+              <>
+                <FormGroup
+                  className="pf-u-mt-md"
+                  fieldId="tls_termination"
+                  isRequired
+                >
+                  <CustomDropdown
+                    options={[
+                      { label: 'Edge', value: 'edge' },
+                      { label: 'Passthrough', value: 'passthrough' },
+                      { label: 'Re-encrypt', value: 're-encrypt' },
+                    ]}
+                    value={formData.conditional.termination}
+                    onChange={(value) => {
+                      updateFormData((prev) => ({
+                        ...prev,
+                        conditional: {
+                          ...prev.conditional,
+                          termination: value,
+                        },
+                      }));
+                    }}
+                    placeholder={t('select_termination')}
+                  />
```

Comment:
Can we move it to another sub-component?
https://github.com/snapp-incubator/contour-console-plugin/pull/2#issuecomment-2591897860